ebill_paynet/models/ebill_payment_contract.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class EbillPaymentContract(models.Model):
+    _inherit = "ebill.payment.contract"
+
+    paynet_account_number = fields.Char(string="Paynet ID", size=20)
+    is_paynet_contract = fields.Boolean(
+        compute='_compute_is_paynet_contract', store=False
+    )
+    paynet_service_id = fields.Many2one(
+        comodel_name="paynet.service",
+        string="Paynet Service",
+        ondelete="restrict",
+    )
+
+    @api.depends('transmit_method_id')
+    def _compute_is_paynet_contract(self):
+        for record in self:
+            record.is_paynet_contract = (
+                record.transmit_method_id
+                == self.env.ref('ebill_paynet.paynet_transmit_method')
```

Comment:
Nitpick: maybe put the call to `ref` outside the `for` loop to get the comparison on one line.

Reply:
Yes changed